tracing/src/main/java/com/palantir/remoting3/tracing/Tracer.java

```diff
@@ -48,8 +52,8 @@ private Tracer() {}
         return trace;
     });
 
-    // Thread-safe Map implementation
-    private static final Map<String, SpanObserver> observers = Maps.newConcurrentMap();
+    private static final Map<String, SpanObserver> observers = new HashMap<>();
+    private static List<SpanObserver> observersList = ImmutableList.of();
```

Comment:
`observersList` needs to be volatile as we're currently only inducing barrier on write side but not reads. overhead of volatile read should be fine on x86

Reply:
To me it's surprising that we'd expect people to register subscribers asynchronously

Reply:
I went with the convention of this class, which is to not do this (e.g. the sampling mode) - but sure, will fix both cases.

Reply:
and yeah, looks like volatile reads are very cheap in x86 provided little contention (which we don't expect here).